src/types/JobParameters.ts

```diff
@@ -1,19 +1,18 @@
-import { FilterQuery, ObjectId } from 'mongodb';
+import type { FilterQuery, ObjectId } from 'mongodb';
 
 export interface IJobParameters<DATA = any | void> {
 	_id?: ObjectId;
 
 	name: string;
 	priority: number;
-	nextRunAt: Date | null;
+	nextRunAt?: Date | null;
 	/**
-	 * // once: the job is just queued in the database --> this does not really exists, it's just  fallback
 	 * normal: job is queued and will be processed (regular case when the user adds a new job)
 	 * single: job with this name is only queued once, if there is an exisitn gentry in the database, the job is just updated, but not newly inserted (this is used for .every())
 	 */
-	type: /* 'once' | */ 'normal' | 'single';
+	type: 'normal' | 'single';
 
-	lockedAt?: Date;
+	lockedAt?: Date | null;
```

Comment:
lockedAt should never be null, it is "unset" in database (=undefiend) when unlocked. or is there a case where lockedAt is set to null?

Reply:
This is for the {$eq: null} comparisons in JobDbRespository.ts

Reply:
also lockJob filters with lockedAt set to null

Reply:
I understand, but this has a special meaning in mongodb. If you query for null it means: property does not exists OR is null OR is undefined.
Therefore I would remove null from here and just add null to the two places for mongodb.
e.g. `const criteria: FilterQuery<IJobParameters & { lockedAt?: Date | null } > `